### PR TITLE
feat: Configure Open VSX registry URL

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
feat: Configure Open VSX registry URL

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21598

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: api/v2/checluster_types.go [hunk 1]
```diff
@@ -242,6 +242,9 @@ type PluginRegistry struct {
 	// External plugin registries.
 	// +optional
 	ExternalPluginRegistries []ExternalPluginRegistry `json:"externalPluginRegistries,omitempty"`
+	// Open VSX registry URL.
```
Comment (l0rd) [inline api/v2/checluster_types.go:245]:
```suggestion
	// Open VSX registry URL. If omitted an embedded instance will be used.
	// To use the Eclipse Foundation operated online service set to `https://openvsx.org`
```

**Review by l0rd — APPROVED** [review]

**Comment by benoitf** [comment]
I'm wondering if we need 'Registry' in the name of the field `OpenVSXRegistryURL` or just `OpenVSXURL` ?
but no strong opinion on both choices

**Comment by l0rd** [comment]
> I'm wondering if we need 'Registry' in the name of the field `OpenVSXRegistryURL` or just `OpenVSXURL` ? but no strong opinion on both choices

Yeah maybe that may be better since we already have the `registry` word in the `pluginRegistry`.

**Comment by tolusha** [comment]
@benoitf 
`devworkspace-happy-path` is very unstable. I always tries to investigate logs to find out the number of failed tests.
It is usually like:
```
  19 passing (10m)
  1 failing
```  
So, I count it as good to merge if all other PR checks passed.

**Pull request merged**